dot/state/epoch.go

```diff
@@ -365,3 +446,89 @@ func (s *EpochState) SkipVerify(header *types.Header) (bool, error) {
 
 	return false, nil
 }
+
+// StoreBABENextEpochData stores the types.NextEpochData under epoch and hash keys
+func (s *EpochState) StoreBABENextEpochData(epoch uint64, hash common.Hash, val types.NextEpochData) {
+	s.nextEpochLock.Lock()
+	defer s.nextEpochLock.Unlock()
+
+	_, has := s.nextEpochData[epoch]
+	if !has {
+		s.nextEpochData[epoch] = make(map[common.Hash]types.NextEpochData)
+	}
+	s.nextEpochData[epoch][hash] = val
+}
+
+// StoreBABENextConfigData stores the types.NextConfigData under epoch and hash keys
+func (s *EpochState) StoreBABENextConfigData(epoch uint64, hash common.Hash, val types.NextConfigData) {
+	s.nextEpochLock.Lock()
+	defer s.nextEpochLock.Unlock()
+
+	_, has := s.nextConfigData[epoch]
+	if !has {
+		s.nextConfigData[epoch] = make(map[common.Hash]types.NextConfigData)
+	}
+	s.nextConfigData[epoch][hash] = val
+}
+
+// FinalizeBABENextEpochData retrieves the types.NextEpochData by epoch and hash keys
+// and delete all the entries for the epoch
+func (s *EpochState) FinalizeBABENextEpochData(epoch uint64) error {
+	s.nextEpochLock.RLock()
+	defer s.nextEpochLock.RUnlock()
+
+	epochData, has := s.nextEpochData[epoch]
+	if !has {
+		return nil
+	}
+
+	for hash, nextEpochData := range epochData {
+		persisted, err := s.blockState.HasHeaderInDatabase(hash)
+		if err != nil {
+			return fmt.Errorf("failed to check header exists at database: %w", err)
+		}
+
+		if !persisted {
+			continue
+		}
```

Comment:
maybe add a comment here explaining what's happening since it's not obvious, if the header is in the database then it's been finalized and thus we can also set the corresponding EpochData in the database